client/my-sites/plans/jetpack-plans/controller.tsx

```diff
@@ -87,6 +89,7 @@ export function jetpackFreeWelcome( context: PageJS.Context, next: () => void ):
 export const jetpackStoragePricing = ( context: PageJS.Context, next: () => void ) => {
 	const { site, duration } = getParamsFromContext( context );
 	const urlQueryArgs: QueryArgs = context.query;
+	const { locale } = context.params;
```

Comment:
`locale` is `undefined` here when I try to load a URL such as: 
http://jetpack.cloud.localhost:3001/fr/pricing/storage/odd-shrew.jurassic.ninja

I think this is because this URL redirects to:
http://jetpack.cloud.localhost:3001/pricing/storage/odd-shrew.jurassic.ninja

Somehow it keeps the french language on the page, but loses the `context.params` value because the language is removed from the URL. Due to this, the buttons pointing to the checkout page are not including the `/fr` locale.

One solution would be to implement the `useLocale()` hook from `@automattic/i18n-utils` inside of the `<StoragePricing />` component instead of passing it in from `context.params`. 

It might also make sense to update this page to keep the language in the URL, like we did with the main `/pricing` page, and then the `context.params` will have the value we would expect to have 👍

Reply:
Instead of getting the locale from `context.params` which only looks at the current URL, get it from the `i18n` object itself, which returns the current locale no matter how it was set -- from URL or anything else:
```js
function SelectorPage() {
  const { locale } = useTranslate();
  buildCheckoutUrl( locale );
}
```
or with a HOC:
```js
function SelectorPage( { locale /* injected by `localize` HOC */ } ) {
  buildCheckoutUrl( locale );
}
export default localize( SelectorPage );
```
That should fix the issues that @nateweller reports.

Reply:
> I have a small concern that it will actually retrieve current locale, which means that redirecting to checkout page will always include language param 

Yes, this is indeed what would happen: the page is always in _some_ language, and the checkout URL would include that locale slug. If the desired behavior is "include language param in the link only if the current page includes a language param, too", then reading from `context.params` is right after all :+1:

Reply:
Thank you and thanks @nateweller for finding the issue. Will try to include all other fixes and fix current behaviour.